.github/workflows/test-action.yml

```diff
@@ -0,0 +1,161 @@
+name: test-sigstore
+
+on:
+  pull_request:
+
+jobs:
+  test_sigstore_action:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      issues: none
+      packages: none
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
+
+    name: Install
+    steps:
+    - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+    - name: Install sigstore gang
+      uses: ./
+    - name: Check cosign install!
+      run: cosign version
+    - name: Check rekor-cli install!
+      run: rekor-cli version
+    - name: Check root directory
+      run: |
+        if [[ $(git diff --stat) != '' ]]; then
+          echo 'should be clean'
+          exit 1
+        else
+          exit 0
+        fi
+      shell: bash
+
+  test_sigstore_action_cosign_custom:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+    permissions:
+      actions: none
+      checks: none
+      contents: none
+      deployments: none
+      issues: none
+      packages: none
+      pull-requests: none
+      repository-projects: none
+      security-events: none
+      statuses: none
+    name: Install Custom Cosign and test presence in path
+    steps:
+    - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+    - name: Install sigstore gang
+      uses: ./
+      with:
+        cosign-version: 'v0.5.0'
+    - name: Check install!
+      run: cosign version
```

Comment:
We could take this a step further and actually compare output of version with the requested version. Somethink like this:

```
COSIGN_VERSION_OUTPUT=$(cosign version 2>&1 /dev/null)
ACTUAL_VERSION=$(echo "$COSIGN_VERSION_OUTPUT" | grep GitVersion | rev | cut -d ' ' -f1 | rev)
```

Reply:
done